syntax = "proto3";

package compiler_gpp;

message Request {
  string request_id = 1;
  int32 sub_id = 2;
  int32 prob_id = 3;
  string file_provider = 4;
  repeated string sub_src_filename = 5;
  repeated string sub_header_filename = 6;
  repeated string prob_src_filename = 7;
  repeated string prob_header_filename = 8;
}

message Response {
  string request_id = 1;
  ResultType result = 2;
  string detail = 3;
  string file_provider = 4;

  enum ResultType {
    SUCCESS = 0;
    FAIL = 1;
    SYSTEM_ERROR = 2;
  }
}